### PR TITLE
fix(DB/SAI): Rallying Cry of the Dragonslayer should happen…

### PR DESCRIPTION
**Pull request by temperrr**
… on quest hand in.

## Changes Proposed:
-  When handing in head of Onyxia (lv 60 q; unavailble without mod-progression-module) or head of nefarian (horde), rallying cry of the dragonslayer should happen and the head on the stick should show

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/11267
- Closes https://github.com/chromiecraft/chromiecraft/issues/3073

## SOURCE:

Vmangos for coordinates.

## Tests Performed:

- Tested

## How to Test the Changes:

1. .q add 7784 as horde
2. hand in and get buff.
3. see the head on the stick
4. notice the two yell lines from the npc

## Known Issues and TODO List:

- [x] Handing in both Head of Onyxia and Head of Nefarian as horde will cause only 1 to happen... -> fixed with conditions and some changes to the smartscript

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Maelthyrr** [comment]
Tested and  work. 

WoWScrnShot_050522_182323

**Comment by Kitzunu** [comment]
iirc that buff should be zone wide. Does that happen with the current script?

**Comment by temperrr** [comment]
image

**Review by Nefertumm — APPROVED** [review]

**Review by Kitzunu — DISMISSED** [review]
Remove quotes from Ints.

Compact your queries

Does this buff apply to all players in the zone. Because iirc that is how it should work.

**Comment by temperrr** [comment]
> Remove quotes from Ints.
> 
> Compact your queries
> 
> Does this buff apply to all players in the zone. Because iirc that is how it should work.

Quotes around its is how Keira generates it. If we don't accept that then we gotta update Keira with this.

Yes this buff applies to all players in the zone like I told you before.

**Comment by Kitzunu** [comment]
AC never allowed it.

**Comment by Kitzunu** [comment]
You never told me anything, just two screenshots that didnt say much. Anyhow, please compact your queries too

File: data/sql/updates/pending_db_world/rev_1651406128820553524.sql [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+(14392, 1, 0, 'Bear witness to the undeniable power of your Warchief! Be lifted by the rallying cry of your dragon slayers!', 14, 0, 100, 0, 0, 0, 9492, 0, 'overlord runthak'),
+(14392, 2, 0, 'NEFARIAN IS SLAIN! People of Orgrimmar, bow down before the might of $n and $g his:her; allies for they have laid a blow against the Black Dragonflight that is sure to stir the Aspects from their malaise! This defeat shall surely be felt by the father of the Black Flight: Deathwing reels in pain and anguish this day!', 14, 0, 100, 0, 0, 0, 9867, 0, 'overlord runthak'),
+(14392, 3, 0, 'Be lifted by $n\'s accomplishment! Revel in $g his:her; rallying cry!', 14, 0, 100, 0, 0, 0, 9868, 0, 'overlord runthak');
+-- Add major mattingly and rhuntak smartai
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` IN (14394, 14392);
+DELETE FROM `smart_scripts` WHERE `source_type` = 0 AND `entryorguid` IN (14394, 14392) OR `source_type` = 9 AND `entryorguid` IN (1439400, 1439200, 1439201);
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/rev_1651406128820553524.sql:12]:
```suggestion
DELETE FROM `smart_scripts` WHERE (`source_type` = 0 AND `entryorguid` IN (14394, 14392)) OR (`source_type` = 9 AND `entryorguid` IN (1439400, 1439200, 1439201)); 
```

**Pull request merged**